display.py

```diff
@@ -30,22 +29,33 @@ def init_ui(self):
 
         for i in range(2):
             for j in range(2):
-                leg = Leg(offset_x + 200 * i - self.offset_3d_x * j, offset_y + self.offset_3d_y * j, leg_length, 0, tibia_length, 0,
-                          j)
+                leg = Leg(offset_x + 200 * i - self.offset_3d_x * j, offset_y + self.offset_3d_y * j, thight_length, 0,
+                          tibia_length, 0,j)
                 legs.append(leg)
                 index += 1
 
         self.canvas.pack(fill=BOTH, expand=1)
 
-    def redraw(self, delete_all=False, draw_trajectory=True):
-        self.canvas.delete("delete")
-        if delete_all:
-            self.canvas.delete("all")
+    # Displays the steps with the trajectory and the step of each joints (with the option of drawing in 3d)
+    def display_step(self, step, draw_trajectory=True, draw_3d=True):
+        """
+        @param step: steps array
+        @param draw_trajectory: boolean option for trajectory visualization
+        @param draw_3d: boolean option for 3D visualization
+        """
+
+        for i, leg_angles in enumerate(step):
+            legs[i].set_angles(leg_angles[0], leg_angles[1])
+
+        # Delete tags before the redraw (display trajectory or no)
```

Comment:
unclear comment

Reply:
> unclear comment

Why is it unclear @turcottep

Reply:
ça dit pas pourquoi on delete quoi dans quel cas

Reply:
It doesn't why delete what parts